spec/API_specification/array_api/array_object.py

```diff
@@ -224,19 +224,40 @@ def __array_namespace__(self: array, /, *, api_version: Optional[str] = None) ->
 
     def __bool__(self: array, /) -> bool:
         """
-        Converts a zero-dimensional boolean array to a Python ``bool`` object.
+        Converts a zero-dimensional array to a Python ``bool`` object.
+
+        .. note::
+           When casting a real-valued input array to ``bool``, a value of ``0`` must cast to ``False``, and a non-zero value must cast to ``True``.
 
         Parameters
         ----------
         self: array
-            zero-dimensional array instance. Must have a boolean data type.
+            zero-dimensional array instance.
 
         Returns
         -------
         out: bool
             a Python ``bool`` object representing the single element of the array.
         """
 
+    def __complex__(self: array, /) -> complex:
+        """
+        Converts a zero-dimensional array to a Python ``complex`` object.
+
+        .. note::
+           When casting a boolean input array, a value of ``True`` must cast to ``1``, and a value of ``False`` must cast to ``0``.
```

Comment:
Would be better to write `1+0j` and `0+0j` here. `complex()` raises a TypeError if `__complex__` doesn't return a complex type.